[Bug]: TypeScript Remix app created with `pnpm create @shopify/app` is missing `@shopify/shopify-api` as dependency

### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

App

### Expected behavior

`@shopify/shopify-api` should be added as a default dependency in `package.json` as `app/shopify.server.js` requires `restResources` from it.

### Actual behavior

`@shopify/shopify-api` is not included as a dependency by default.

### Verbose output

```shell
Cannot find module '@shopify/shopify-api/rest/admin/2023-10' or its corresponding type declarations.ts(2307)
```


### Reproduction steps

1. Create a new app with `pnpm create @shopify/app`
2. Open `app/shopify.server.ts`
3. Observe that the `restResources` import fails as the required package is not installed

### Operating System

macOS Sonoma

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.50.0

### Shell

zsh

### Node version (run `node -v` if you're not sure)

v20.8.0

### What language and version are you using in your application?

Node 20

--- a/package.json
+++ b/package.json
@@ -27,6 +27,7 @@
     "@shopify/app-bridge-types": "^0.0.3",
     "@shopify/cli": "^3.50.0",
     "@shopify/polaris": "^12.0.0",
+    "@shopify/shopify-api": "^8.0.2",
     "@shopify/shopify-app-remix": "^2.1.0",
     "@shopify/shopify-app-session-storage-prisma": "^2.0.0",
     "cross-env": "^7.0.3",

